recommender/migrate/utils.py

```diff
@@ -0,0 +1,156 @@
+# pylint: disable=invalid-name
+
+"""Util methods for migration purposes"""
+
+import os
+from importlib import import_module
+
+from definitions import MIGRATE_DIR
+from recommender.models.migration import Migration
+
+
+def _get_migration_class(migration: Migration):
+    migration_name = migration.name
+    capitalized = [part.capitalize() for part in migration_name.split("_")[1:]]
+    class_name = "".join(capitalized)
+
+    migration_module = import_module(f"recommender.migrate.{migration_name}")
+    migration_class = getattr(migration_module, class_name)
+
+    return migration_class
+
+
+def _get_migrations_from_migrate_dir():
+    files = os.listdir(MIGRATE_DIR)
+    migration_names = [
+        file.split(".")[0] for file in files if file.split("_")[0].isnumeric()
+    ]
+    return migration_names
+
+
+def scan_for_migrations():
+    """
+    Scans for migrations in the MIGRATE_DIR and
+    then inserts newly found migrations into the Migration collection.
+    """
+
+    print("Scanning for new migrations...")
+
+    all_migration_names = _get_migrations_from_migrate_dir()
+    existing_migration_names = [migration.name for migration in Migration.objects()]
+    missing_migration_names = set(all_migration_names) - set(existing_migration_names)
+
+    if missing_migration_names:
+        for m in missing_migration_names:
+            print(f"\tInserting new migration {m} into the Migration collection...")
+            new_migration_model = Migration(name=m, applied=False)
+            new_migration_model.save()
+
+        print("\tNew migrations acknowledged!")
+        return True
+
+    print("\tFound no new migrations!")
+    return False
+
+
+def apply_migrations():
+    """Applies unapplied migrations"""
+
+    print("Applying migrations...")
+
+    unapplied_migrations = Migration.objects(applied=False).order_by("+name")
+
+    if unapplied_migrations:
+        for migration in unapplied_migrations:
+            print(f"\tApplying `{migration.name}` ...")
+
+            # Retrieve class
+            migration_class = _get_migration_class(migration)
+
+            # Apply migration
+            migration_class().up()
+
+            # Update migration document
+            migration.update(applied=True)
+
+        print("\tDone applying migrations!")
+        return True
+
+    print("\tNo unapplied migrations found!")
+    return False
+
+
+def rollback_last_migration():
+    """Rollbacks last applied migration"""
+
+    print("Rollbacking last applied migration...")
+
+    applied_migrations = Migration.objects(applied=True).order_by("-name")
+
+    if applied_migrations:
+        last_applied = applied_migrations.first()
+        print(f"\tRollbacking `{last_applied.name}` ...")
+
+        # Retrieve class
+        migration_class = _get_migration_class(last_applied)
+
+        # Rollback migration
+        migration_class().down()
```

Comment:
If migration does not implement `down` it should raise error - this would abort the script. It should be good enough (the error could be caught and nice message may be printed).